mrchecker-framework-modules/mrchecker-webapi-module/src/test/java/com/capgemini/mrchecker/webapi/httpbin/RandomNBytesTest.java

```diff
@@ -0,0 +1,41 @@
+package com.capgemini.mrchecker.webapi.httpbin;
+
+import com.capgemini.mrchecker.test.core.logger.BFLogger;
+import com.capgemini.mrchecker.webapi.BasePageWebApiTest;
+import com.capgemini.mrchecker.webapi.pages.httbin.RandomNBytesPage;
+import io.restassured.response.Response;
+import org.junit.Test;
+
+import java.text.MessageFormat;
+import java.util.Random;
+
+import static org.hamcrest.Matchers.is;
+import static org.junit.Assert.assertThat;
+
+public class RandomNBytesTest extends BasePageWebApiTest {
+	private RandomNBytesPage randomNBytesPage = new RandomNBytesPage();
+
+	@Test
+	public void staticBytesTest() {
+		Integer size=10;
+		validateNBytesOfData(size);
+	}
+
+	@Test
+	public void randomBytesTest() {
+		Random generator = new Random();
+		Integer size=generator.nextInt(50);
+		validateNBytesOfData(size);
+	}
+
+	private void validateNBytesOfData(Integer size) {
+		BFLogger.logInfo(MessageFormat.format("Step 1 - Sending GET query to {0} with valid value(size): {1}", randomNBytesPage.getEndpoint(), size));
+		Response response = randomNBytesPage.sendGETQuery(Integer.toString(size));
+
+		BFLogger.logInfo("Step 2 - Validate response status code (should be 200): ");
+		assertThat(response.statusCode(), is(200));
+
+		BFLogger.logInfo(MessageFormat.format("Step 3 - Validate response body (should be {0}): ", size));
+		assertThat(response.body().asByteArray().length, is(size));
```

Comment:
`randomNBytesPage` should provide getter method returning ByteArray
For example
`randomNBytesPage.getRandomBytes(size)`

Reply:
`getRandomBytes` should use `response` instead of `size`, we want to work with the same data, not with a new one every time we run the function

Reply:
`getRandomBytes(int size)` method should just return a `ByteArray` containing response from endpoint of size `size`. You can then save it to variable and do whatever you want. In addition to that, `randomNBytesPage` can provide getter for status code and other needed things. Calling endpoint in order to get random bytes, could also be in another method.
What I want is to make `randomNBytesPage` object helping us in our work with endpoint, not just storing endpoint data and returning `Response` object. We could achieve this without using any page object, just calling methods from RestAssured in a test class.
I hope that my explanation makes sense.

Reply:
Run after review: http://10.40.234.40:8080/job/MrChecker-SDLC/job/CI_mrchecker-webapi-module/89

Reply:
Done